app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/CreateRecordConnector.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.connectors
+
+import com.google.inject.Inject
+import play.api.http.MimeTypes
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.tradergoodsprofilesrouter.config.AppConfig
+import uk.gov.hmrc.tradergoodsprofilesrouter.connectors.EisHttpReader.HttpReader
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.CreateRecordPayload
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.CreateOrUpdateRecordResponse
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.DateTimeService
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.DateTimeService.DateTimeFormat
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.HeaderNames
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CreateRecordConnector @Inject() (
+  appConfig: AppConfig,
+  httpClientV2: HttpClientV2,
+  dateTimeService: DateTimeService
+)(implicit val ec: ExecutionContext)
+    extends EisHttpErrorHandler {
+
+  def createRecord(
+    payload: CreateRecordPayload,
+    correlationId: String
+  )(implicit hc: HeaderCarrier): Future[Either[Result, CreateOrUpdateRecordResponse]] = {
+    val url = appConfig.eisConfig.createRecordUrl
+
+    httpClientV2
+      .post(url"$url")
+      .setHeader(eisRequestHeaders(correlationId, appConfig.eisConfig.createRecordBearerToken): _*)
+      .withBody(Json.toJson(payload))
+      .execute(HttpReader[CreateOrUpdateRecordResponse](correlationId, handleErrorResponse), ec)
+  }
+
+  private def eisRequestHeaders(correlationId: String, bearerToken: String)(implicit
```

Comment:
I think this method could go into it's own service now which all these connectors could extend.

Reply:
Yes, I was thinking to extract it in the net iteration